pkg/util/containerd/containerd_util.go

```diff
@@ -72,37 +70,52 @@ type ContainerdUtil struct {
 // GetContainerdUtil creates the Containerd util containing the Containerd client and implementing the ContainerdItf
 // Errors are handled in the retrier.
 func GetContainerdUtil() (ContainerdItf, error) {
-	once.Do(func() {
-		globalContainerdUtil = &ContainerdUtil{
-			queryTimeout:      config.Datadog.GetDuration("cri_query_timeout") * time.Second,
-			connectionTimeout: config.Datadog.GetDuration("cri_connection_timeout") * time.Second,
-			socketPath:        config.Datadog.GetString("cri_socket_path"),
-			namespace:         config.Datadog.GetString("containerd_namespace"),
-		}
-		if globalContainerdUtil.socketPath == "" {
-			log.Info("No socket path was specified, defaulting to /var/run/containerd/containerd.sock")
-			globalContainerdUtil.socketPath = containerdDefaultSocketPath
-		}
-		// Initialize the client in the connect method
-		globalContainerdUtil.initRetry.SetupRetrier(&retry.Config{ //nolint:errcheck
-			Name:              "containerdutil",
-			AttemptMethod:     globalContainerdUtil.connect,
-			Strategy:          retry.Backoff,
-			InitialRetryDelay: 1 * time.Second,
-			MaxRetryDelay:     5 * time.Minute,
-		})
+	// A singleton does not work because different parts of the code
```

Comment:
Might consider renaming it to `CreateContainerUtil` or `NewContainerdUtil`? The `GetXXXUtil` pattern is used very extensively in the agent to mean you get a singleton instead of creating a new instance.

Reply:
Changed 👍 
Added a new commit with this.